plugin/trino-open-policy-agent/src/main/java/io/trino/plugin/openpolicyagent/OpaAuthorizer.java

```diff
@@ -0,0 +1,827 @@
+    {
+        OpaQueryInputResource resource = new OpaQueryInputResource.Builder().table(new OpaQueryInputResource.Table(table, updatedColumnNames)).build();
+
+        if (!queryOpaWithSimpleResource(securityContext, "UpdateTableColumns", resource)) {
+            SystemAccessControl.super.checkCanUpdateTableColumns(securityContext, table, updatedColumnNames);
+        }
+    }
+
+    @Override
+    public void checkCanCreateView(SystemSecurityContext context, CatalogSchemaTableName view)
+    {
+        OpaQueryInputResource resource = new OpaQueryInputResource.Builder().view(new OpaQueryInputResource.Table(view)).build();
+        if (!queryOpaWithSimpleResource(context, "CreateView", resource)) {
+            SystemAccessControl.super.checkCanCreateView(context, view);
+        }
+    }
+
+    @Override
+    public void checkCanRenameView(SystemSecurityContext context, CatalogSchemaTableName view, CatalogSchemaTableName newView)
+    {
+        OpaQueryInputResource resource = new OpaQueryInputResource.Builder().view(new OpaQueryInputResource.Table(view)).build();
+        OpaQueryInputResource targetResource = new OpaQueryInputResource.Builder().view(new OpaQueryInputResource.Table(newView)).build();
+        OpaQueryInputAction action = new OpaQueryInputAction.Builder().operation("RenameView").resource(resource).targetResource(targetResource).build();
+        OpaQueryInput input = new OpaQueryInput(context, action);
+
+        if (!queryOpa(input)) {
+            SystemAccessControl.super.checkCanRenameView(context, view, newView);
+        }
+    }
+
+    @Override
+    public void checkCanSetViewAuthorization(SystemSecurityContext context, CatalogSchemaTableName view, TrinoPrincipal principal)
+    {
+        OpaQueryInputResource resource = new OpaQueryInputResource.Builder().view(new OpaQueryInputResource.Table(view)).build();
+        OpaQueryInputGrant grantee = new OpaQueryInputGrant.Builder().principal(principal).build();
+        OpaQueryInputAction action = new OpaQueryInputAction.Builder().operation("SetViewAuthorization").resource(resource).grantee(grantee).build();
+        OpaQueryInput input = new OpaQueryInput(context, action);
+
+        if (!queryOpa(input)) {
+            SystemAccessControl.super.checkCanSetViewAuthorization(context, view, principal);
+        }
+    }
+
+    @Override
+    public void checkCanDropView(SystemSecurityContext context, CatalogSchemaTableName view)
+    {
+        OpaQueryInputResource resource = new OpaQueryInputResource.Builder().view(new OpaQueryInputResource.Table(view)).build();
+        if (!queryOpaWithSimpleResource(context, "DropView", resource)) {
+            SystemAccessControl.super.checkCanDropView(context, view);
+        }
+    }
+
+    @Override
+    public void checkCanCreateViewWithSelectFromColumns(SystemSecurityContext context, CatalogSchemaTableName table, Set<String> columns)
+    {
+        // This refers to a Table, so the resource should be a table and not a view
```

Comment:
A view can query another view. Why is there a distinction between views and tables anyway? In the SQL model they are meant to be basically indistinguishable.

Reply:
Agreed - and they pretty much are within the code, they both use the `Table` object. I will change this so that the resource is always `table`